cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -223,6 +239,8 @@ def _control_keys_(self) -> AbstractSet['cirq.MeasurementKey']:
                 if not protocols.control_keys(self.circuit)
                 else protocols.control_keys(self.mapped_circuit())
             )
+            if self.repeat_until is not None:
```

Comment:
This should be removed. We define control keys as things that must be defined prior to the operation. That would be true for a "do-while", but a "repeat-until" does not depend on anything prior to the subcircuit itself. It depends only on the measurement occurring within the subcircuit, which is checked in `_post_init_`.

Reply:
This prevents something like:
```
measure q0, key=k1
circuit operation:
  measure q1, key=k2
  repeat until k1 == k2
```
Is there a low-cost way to allow this behavior? I don't have a specific use case in mind, but this seems like the sort of edge case that a user would be surprised to find doesn't work.

Reply:
I think this should work already with sympy. Will write a test when I get home.

Reply:
It works in simulation, but `k1` won't appear in its control keys, possibly leading to misbehavior during construction.

Reply:
I see, yeah and commutation assumptions would be broken. Will fix this up when I get home.

Reply:
Fixed (still slightly different from the original implementation--only *unbound* keys (k1 in your example) are added to the control_keys response), and added test. Good catch!